src/compiler/sys.ts

```diff
@@ -1072,11 +1083,12 @@ namespace ts {
                         }
                     }
                     try {
-
                         const presentWatcher = _fs.watch(
                             fileOrDirectory,
                             options,
-                            callback
+                            isLinuxOrMacOs ?
+                                callbackChangingToMissingFileSystemEntry :

```

Comment:
So this callback can change to the Missing FileSystemEntry from the Present one, right? If so, why not always use it?

How do we detect missing files -- is it when relativeName is incorrect as handled below? Is it possible for relativeName to be incorrect in other ways that don't indicate missing files?

Reply:
The missing file system entry is watched by polling on the directory stat so it is cpu extensive and we don't want to do that for directories. 
The name is incorrect in a way it is undefined or doesn't contain full path just base file name. If you look at the check, it is more permissive to check directory existence. So node_modules in child folder will also trigger check but that is more accurate than not checking in that scenario.